src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -123,12 +125,17 @@ public function supportsClass($class)
     protected function loadUser($username, Entry $entry)
     {
         $password = null;
+        $metadata = [];
 
         if (null !== $this->passwordAttribute) {
             $password = $this->getAttributeValue($entry, $this->passwordAttribute);
         }
 
-        return new User($username, $password, $this->defaultRoles);
+        foreach ($this->userMetadata as $field) {
```

Comment:
No need for checking if the field exists?

Reply:
It already does the check in the `getAttributeValue` method, we don't check anything else because there no real field in the User::class.